scheduler/src/test/java/org/apache/mesos/elasticsearch/scheduler/ConfigurationTest.java

```diff
@@ -63,19 +64,148 @@ public void shouldCreateArguments() {
         final ClusterState clusterState = Mockito.mock(ClusterState.class);
         final int port = 1234;
         final Protos.DiscoveryInfo discoveryInfo = Protos.DiscoveryInfo.newBuilder().setPorts(Protos.Ports.newBuilder()
-                .addPorts(Protos.Port.newBuilder().setNumber(port))
-                .addPorts(Protos.Port.newBuilder().setNumber(port)))
-                .setVisibility(Protos.DiscoveryInfo.Visibility.EXTERNAL)
-                .build();
+          .addPorts(Protos.Port.newBuilder().setNumber(port))
+          .addPorts(Protos.Port.newBuilder().setNumber(port)))
+          .setVisibility(Protos.DiscoveryInfo.Visibility.EXTERNAL)
+          .build();
         Protos.SlaveID slaveID = Protos.SlaveID.newBuilder().setValue("SLAVE").build();
         final List<String> arguments = configuration.esArguments(clusterState, discoveryInfo, slaveID);
         String allArgs = arguments.toString();
         assertTrue(allArgs.contains(Integer.toString(port)));
     }
 
+    @Test
+    public void shouldCreateArgumentsWithHostname() {
+        final String discoveryHostname = "ahostnameidefined";
+        Configuration configuration = new Configuration(ZookeeperCLIParameter.ZOOKEEPER_MESOS_URL, "aa", Configuration.DISCOVERY_ZEN_PING_UNICAST_HOSTS, discoveryHostname);
+        final ClusterState clusterState = Mockito.mock(ClusterState.class);
+        final int port = 1234;
+        final Protos.DiscoveryInfo discoveryInfo = Protos.DiscoveryInfo.newBuilder().setPorts(Protos.Ports.newBuilder()
+          .addPorts(Protos.Port.newBuilder().setNumber(port))
+          .addPorts(Protos.Port.newBuilder().setNumber(port)))
```

Comment:
Nit: I'm not familiar with DiscoveryInfo... but do you need to add multiple ports here?

Reply:
Oddly enough it does (as strange as this looks). I thought the same thing when I wrote the test.

Needs ports for client and transport, and uses indices set in `Discovery` class to grab them out of `DiscoveryInfo`. Test ends in a NPE if you don't set two ports here.
```   public class Discovery {
    // An elasticsearch task has 2 ports, client port and transport port.
    public static final int CLIENT_PORT_INDEX = 0;
    public static final int TRANSPORT_PORT_INDEX = 1;
    public static final String CLIENT_PORT_NAME = "CLIENT_PORT";
    public static final String TRANSPORT_PORT_NAME = "TRANSPORT_PORT";
    public static final int EXPECTED_NUMBER_OF_PORTS = 2;
}
```